scripts/create-functionmap.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+$filter = function (string $name): bool {
+    $namespaces = ['MongoDB\BSON\\', 'MongoDB\Driver\\'];
+
+    foreach ($namespaces as $namespace) {
+        if (str_starts_with($name, $namespace) || str_starts_with($name, strtolower($namespace))) {
+            return true;
+        }
+    }
+
+    return false;
+};
+
+$getFunctionCallMapEntry = function(ReflectionFunctionAbstract $function): array {
+    $returnType = match(true) {
```

Comment:
> This may be one of the first scripts to require PHP 8+.

Almost: gen_stub.php also requires the latest PHP version by default. Either way, I'll add a note about PHP 8 in a comment above, but as you already suggested the `match` statement will fail to parse.